### PR TITLE
Improving format compat strip

### PR DESCRIPTION
**Pull request by mmatera**
This is the second part of #302

<a href="https://gitpod.io/#https://github.com/Mathics3/mathics-core/pull/307"></a>

### CODE REVIEW

**Comment by mmatera** [comment]
@rocky, @TiagoCavalcante I think this is nos ready for review.

File: mathics/builtin/box/inout.py [hunk 1]
```diff
@@ -465,41 +477,38 @@ def apply_list(self, boxes, evaluation):
...
-                elements.append(item.to_expression())
-            else:
-                elements.append(item)
-        elements = Expression(SymbolList, *elements)
-        self._elements = tuple((elements,))
+        self._elements = None
 
     def to_expression(self):
         """
         returns an evaluable expression.
```
Comment (rocky) [inline mathics/builtin/box/inout.py:504]: A very small thing, but I'd appreciate if we didn't evaluable which isn't a word. 

If this were:

```
to_expression(self) -> Expression: 
```

That would be saying exactly the same thing, if a form that Python programmers should understand better.

Yeah, I know this isn't in the part that was changed.

File: mathics/builtin/box/inout.py [hunk 2]
```diff
@@ -465,41 +477,38 @@ def apply_list(self, boxes, evaluation):
...
         """
-        items = (item.to_expression() for item in self.items)
-        result = Expression("RowBox", Expression("List", *items))
+        if self._elements is None:
+            self._elements = tuple(
+                item.to_expression() if isinstance(item, BoxExpression) else item
+                for item in self.items
+            )
+        result = Expression("RowBox", Expression("List", *self._elements))
```
Comment (rocky) [inline mathics/builtin/box/inout.py:511]: Will be converting `Expresssion("List",...)` and `Expression(SymboList,...)` kinds of things soon, so there isn't a need for tthis to get changed.

I'll probably also throw in `SymbolRowBox` as well then too.

File: mathics/core/atoms.py [hunk 3]
```diff
@@ -812,6 +796,10 @@ def boxes_to_text(self, show_string_characters=False, **options) -> str:
 
         return value
 
+    # ``boxes_to_*`` are going to be removed from the ``Atom`` interface.
```
Comment (rocky) [inline mathics/core/atoms.py:799]: Removing `boxes_to_*`  and replacing it with a more extensible mechanism would be awesome!

File: mathics/core/atoms.py [hunk 4]
```diff
@@ -812,6 +796,10 @@ def boxes_to_text(self, show_string_characters=False, **options) -> str:
...
+    # For ``String`` however, it is still tricky to move all to _BoxedString.
+    # Another option would be to unify BoxedString and String, and make that both
+    # be BoxExpressions.
```
Comment (rocky) [inline mathics/core/atoms.py:802]: I agree, I think BoxExpression might be the way to go.

**Review by rocky — APPROVED** [review]
LGTM - thanks for doing. See comments if you are up for adjsting.

File: mathics/core/atoms.py (hunk 3 above)
Comment (mmatera) [reply mathics/core/atoms.py:799]: With the last changes, we can remove all these methods, except for the ones in String. Once this get merged, that is the next step

**Pull request merged**